stitch/src/androidTest/java/com/mongodb/stitch/PushTests.kt

```diff
@@ -0,0 +1,158 @@
+package com.mongodb.stitch
+
+import android.app.Instrumentation
+import android.content.Context
+import android.content.Intent
+import android.support.test.rule.ServiceTestRule
+import android.support.test.runner.AndroidJUnit4
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.android.gms.common.api.GoogleApi
+import com.mongodb.stitch.admin.services.ServiceConfigWrapper
+import com.mongodb.stitch.admin.services.ServiceConfigs
+import com.mongodb.stitch.android.push.PushProviderName
+import com.mongodb.stitch.android.push.gcm.GCMPushClient
+import com.mongodb.stitch.android.push.gcm.GCMPushProviderInfo
+import com.mongodb.stitch.pushTest.AndroidTestGCMListenerService
+import org.bson.Document
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.doReturn
+import org.mockito.MockitoAnnotations
+import kotlin.test.assertFails
+import kotlin.test.assertNotNull
+
+@RunWith(AndroidJUnit4::class)
+class PushTests: StitchTestCase() {
+    private val senderId = "405021717222"
+    @Mock
+    var mockGoogleApiAvailability: GoogleApiAvailability? = null
+
+    @get:Rule
+    val mServiceRule = ServiceTestRule()
+
+    @Before
+    override fun setup() {
+        super.setup()
+
+        System.setProperty("dexmaker.dexcache", instrumentationCtx.cacheDir.path)
+        MockitoAnnotations.initMocks(this)
+        await(harness.add(ServiceConfigWrapper(
+                name = "gcm",
+                type = "gcm",
+                config = ServiceConfigs.GCM(
+                        senderId = senderId,
+                        apiKey = "AIzaSyDLW2UlIbT6wZQev8x_tUX_YlHZIECrzZc"
```

Comment:
[minor] Are we fine exposing this? I know it's probably a dummy app, but I'm not entirely sure what the implications are or if it could affect our API quotas/billing if someone abused it. If there's a way to do this with env variables and evergreen that would probably be better.